### PR TITLE
Fix options passing in Json and Attributes adapters

### PR DESCRIPTION
**Pull request by beauby**
As mentioned in #1184, there are issues with the way options are passed between adapters. This PR adds a failing test and provides fixes.

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
:+1:

File: test/adapter/json/collection_test.rb [hunk 1]
```diff
@@ -75,6 +75,16 @@ def test_root_is_underscored
 
             assert_equal 1, adapter.serializable_hash[:virtual_values].length
           end
+
+          def test_include_option
+            serializer = ArraySerializer.new([@first_post, @second_post])
+            adapter = ActiveModel::Serializer::Adapter::Json.new(serializer, include: '')
```
Comment (bf4) [inline test/adapter/json/collection_test.rb]: what did it do before?

Comment (beauby) [inline test/adapter/json/collection_test.rb]: It disregarded the `include` option.

Comment (beauby) [inline test/adapter/json/collection_test.rb]: It did so every time for the Json adapter, and only for collections for the Attribute adapter.

Comment (bf4) [inline test/adapter/json/collection_test.rb]: Ok, but when it failed, how was the output different? (lazy me)

Comment (beauby) [inline test/adapter/json/collection_test.rb]: It did include stuff, although it was explicitly told not to.

**Pull request merged**